pkg/service/service_test.go

```diff
@@ -484,3 +484,42 @@ func (m *ManagerTestSuite) TestGetServiceNameByAddr(c *C) {
 	_, _, ok = m.svc.GetServiceNameByAddr(frontend2.L3n4Addr)
 	c.Assert(ok, Equals, false)
 }
+
+func (m *ManagerTestSuite) TestLocalRedirectLocalBackendSelection(c *C) {
+	// Create a node-local backend
+	localBackend := *lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.1"), 8080)
+	localBackend.NodeName = nodeTypes.GetName()
+	localBackends := []lb.Backend{localBackend}
+	// Create two remote backends
+	remoteBackend1 := *lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.3"), 8080)
+	remoteBackend1.NodeName = "not-" + nodeTypes.GetName()
+	remoteBackend2 := *lb.NewBackend(0, lb.TCP, net.ParseIP("10.0.0.4"), 8080)
+	remoteBackend2.NodeName = "not-" + nodeTypes.GetName()
+	allBackends := []lb.Backend{localBackend, remoteBackend1, remoteBackend2}
+
+	// Create a service entry of type Local Redirect
+	p1 := &lb.SVC{
+		Frontend:      frontend1,
+		Backends:      allBackends,
+		Type:          lb.SVCTypeLocalRedirect,
+		TrafficPolicy: lb.SVCTrafficPolicyCluster,
+		Name:          "svc1",
+		Namespace:     "ns1",
+	}
+	// Insert the service entry of type Local Redirect
+	created, id, err := m.svc.UpsertService(p1)
+	c.Assert(err, IsNil)
+	c.Assert(created, Equals, true)
+	c.Assert(id, Not(Equals), lb.ID(0))
+
+	svc, ok := m.svc.svcByID[id]
+	c.Assert(ok, Equals, true)
+	c.Assert(svc.svcNamespace, Equals, "ns1")
+	c.Assert(svc.svcName, Equals, "svc1")
+	// Only node-local backends are selected
+	c.Assert(len(svc.backends), Equals, len(localBackends))
+
```

Comment:
Maybe worth adding a test which checks if an err is returned when a matching SVC exists for a to-be inserted LRP svc?

Reply:
Yes, I plan to add that test. I was waiting to for your review to see if I missed anything else in that regard.